#165 switch formatter from scalariform to scalafmt

fg-devs opened the pull request.

lightbend-cla-validator commented:
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

fg-devs commented:
Refs #165

fg-devs commented:
Hey @ignasi35! Thanks for the feedback.

That is a good idea with splitting the commits, I am gonna change it 👍 

> add scalafmt's sbt plugin. We generally prefer that over running scalafmt's CLI externally.
This is what I actually did. I think this becomes more visible once I split up the commits.

> remove scalariform completely.
I removed the sbt plugin and couldn't find any further usages, so I am not exactly sure where you think it is still used.

> setup .travis.conf to check formatting. This currently is based on an sbt alias that uses scalariform under the covers.
Yeah I have adapted this as well, the alias still exists but used the scalafmt's task instead and not scalariform anymore (probably also becomes more visible when commits are split up)

ignasi35 commented:
> I removed the sbt plugin and couldn't find any further usages, so I am not exactly sure where you think it is still used.

I noticed there as still an `sbtCommandAlias` using `scalariform` and that made me think there should be a compile error. Therefore I assumed scalariform had to be in the project. But that's not correct! Alias are check at compile time.

ignasi35 commented on build.sbt:
```diff
@@ -589,20 +589,4 @@ lazy val PomExtra = {
-addCommandAlias("validateCode",
-  ";scalariformFormat;test:scalariformFormat;checkCodeFormat"
-)
+addCommandAlias("validateCode", ";+scalafmtCheckAll;scalafmtSbtCheck")
```
Hmm, but do we need it? Is there any customization in `scalatestplus-play` which depends on the scala version? (we used cross-validation in `playframework/playframework` I think but is it necessary here?)

fg-devs replied:
Yeah actually true, I am gonna remove it

ignasi35 commented on scripts/validateCode:
```diff
@@ -4,6 +4,11 @@
+    echo "[error] To fix, format your sources using 'sbt scalafmtAll scalafmtSbt' before submitting a pull request."
+    echo "[error] Additionally, please squash your commits (eg, use git commit --amend) if you're going to update this pull request."
+    false
+  )
```
Hmmm, should we get rid of the `validateCode` alias in `build.sbt` then?

fg-devs replied:
Hmm I think we have two choices here, either change it here to use the command alias like `build validateCode || ...` or keep it like that, but then we also have to update the README.md because there is a reference to run `sbt validateCode` before pushing code. I would vote for the command alias as it wraps both scalafmt tasks, what do you think?

ignasi35 replied:
I'm fine either way. As long as we keep things consistent :-)

Keeping `sbt validateCode` probably requires fewer changes.

ignasi35 commented:
Nice! thanks @fg-devs
